scripts/build_connectome/connectivity_dmri.py

```diff
@@ -0,0 +1,442 @@
+    )
+    # transform atlas to diffusion native space
+    atlas2dwi(b0dwi, mni_nifti, atlas.maps, atlas_in_dwi_space)
+
+    ######## start of connectivity matrix creation ########
+    # iterate over spaces
+    for space in ["MNI152", "individual"]:
+        if space == "MNI152":
+            # use tractogram in mni space
+            tck = mni_tck
+            # use original atlas (in mni space)
+            atlas = atlas.maps
+        else:
+            # use original tractogram in diffusion native space
+            tck = dwi_tck
+            # use transformed atlas in diffusion native space
+            atlas = atlas_in_dwi_space
+        # calculate connectivity matrices with and without sift weights
+        for sift in ["siftweighted", "nosift"]:
+            if sift == "siftweighted":
+                # file path for previously calculated sift weights
+                sift_weights = os.path.join(
+                    dwi_dir,
+                    f"sift-track_{sub}_{ses}.txt",
+                )
+            else:
+                sift_weights = None
+            # name for output connectivity matrix
+            connectivity_matrix = os.path.join(
+                dwi_dir,
+                f"{sub}_{ses}_Diffusion_connectome_{atlas.name}_{space}_"
+                f"{sift}.csv",
+            )
+            # name for output inverse connectivity matrix
+            inverse_connectivity_matrix = os.path.join(
+                dwi_dir,
+                f"{sub}_{ses}_Diffusion_invconnectome_{atlas.name}_{space}_"
+                f"{sift}.csv",
+            )
+            # calculate and save the connectivity and inv connectivity
+            # matrices
+            tck2connectome(
+                atlas,
+                tck,
+                connectivity_matrix,
+                inverse_connectivity_matrix,
+                sift_weights=sift_weights,
+            )
+
+    return f"{sub} Done!"
+
+
+if __name__ == "__main__":
+    # cache directory
+    cache = "/storage/store/work/haggarwa/"
+    # get atlas
+    atlas = datasets.fetch_atlas_schaefer_2018(
+        data_dir=cache, resolution_mm=1, n_rois=400
+    )
+    # give atlas a custom name
+    atlas["name"] = "schaefer400"
+    # download mni templates and mask
+    fetch_mni_template(cache)
+    # path to mni t2w template
+    mni_unmasked = os.path.join(
+        cache,
+        "mni_icbm152_nlin_sym_09a",
+        "mni_icbm152_t2_tal_nlin_sym_09a.nii",
+    )
+    # path to mni t1w template mask
+    mni_mask = os.path.join(
+        cache,
+        "mni_icbm152_nlin_sym_09a",
+        "mni_icbm152_t1_tal_nlin_sym_09a_mask.nii",
+    )
+    # path to masked mni t2w
+    mni_nifti = os.path.join(cache, "mni_t2w.nii.gz")
+    # apply mask to mni t2w template
+    apply_mask(mni_unmasked, mni_mask, mni_nifti)
+    # get sessions with diffusion data
+    subject_sessions = sorted(get_subject_session("anat1"))
+    sub_ses = {
+        subject_session[0]: "ses-12"
```

Comment:
This looks weird. Not sure what you're doing here

Reply:
This just creates a dictionary with `sub: ses` as key-value pairs. The output of `get_subject_session` isn't very clean for diffusion data (`anat1`). So I had to use if..else to set the right session numbers for sub-01 and sub-15.